plugins/modules/proxmox_template.py

```diff
@@ -248,14 +298,18 @@ def main():
         timeout=dict(type='int', default=30),
         force=dict(type='bool', default=False),
         state=dict(default='present', choices=['present', 'absent']),
+        checksum_algorithm=dict(choices=['md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512']),
+        checksum=dict(type='str'),
+        verify_checksum=dict(type='bool', default=False)
     )
     module_args.update(template_args)
 
     module = AnsibleModule(
         argument_spec=module_args,
         required_together=[('api_token_id', 'api_token_secret')],
```

Comment:
You might want to add another required together for `checksum` and `checksum_algorithm`.